Neos.Neos/Classes/Fusion/Helper/DimensionHelper.php

```diff
@@ -0,0 +1,202 @@
+     */
+    public function findVariantInDimension(Node $node, string|ContentDimensionId $dimensionName, string|ContentDimensionValue $dimensionValue): ?Node
+    {
+        $contentDimensionId = is_string($dimensionName) ? new ContentDimensionId($dimensionName) : $dimensionName;
+        $contentDimensionValue = is_string($dimensionValue) ? new ContentDimensionValue($dimensionValue) : $dimensionValue;
+        $contentRepository = $this->contentRepositoryRegistry->get($node->subgraphIdentity->contentRepositoryId);
+
+        return $contentRepository
+            ->getContentGraph()
+            ->getSubgraph($node->subgraphIdentity->contentStreamId, $node->subgraphIdentity->dimensionSpacePoint->vary($contentDimensionId, $contentDimensionValue->value), $node->subgraphIdentity->visibilityConstraints)
+            ->findNodeById($node->nodeAggregateId);
+    }
+
+    /**
+     * Find all variants of `node` across the specified dimension.
+     *
+     * Example::
+     *
+     *     Neos.Dimension.findVariantsInDimension(node, 'language')
+     *
+     * @param Node $node
+     * @param string|ContentDimensionId $dimensionName
+     * @return Nodes
+     */
+    public function findVariantsInDimension(Node $node, string|ContentDimensionId $dimensionName): Nodes
+    {
+        $contentDimensionId = is_string($dimensionName) ? new ContentDimensionId($dimensionName) : $dimensionName;
+        $contentRepository = $this->contentRepositoryRegistry->get($node->subgraphIdentity->contentRepositoryId);
+
+        $variantNodes = [];
+        foreach ($contentRepository->getVariationGraph()->getDimensionSpacePoints() as $dimensionSpacePoint) {
+            if ($dimensionSpacePoint->equals($node->subgraphIdentity->dimensionSpacePoint)) {
+                $variantNodes[] = $node;
+            } elseif (
+                $dimensionSpacePoint->isDirectVariantInDimension(
+                    $node->subgraphIdentity->dimensionSpacePoint,
+                    $contentDimensionId
+                )
+            ) {
+                $variantNode = $contentRepository
+                    ->getContentGraph()
+                    ->getSubgraph($node->subgraphIdentity->contentStreamId, $dimensionSpacePoint, $node->subgraphIdentity->visibilityConstraints)
+                    ->findNodeById($node->nodeAggregateId);
+
+                if ($variantNode instanceof Node) {
+                    $variantNodes[] = $variantNode;
+                }
+            }
+        }
+
+        return Nodes::fromArray($variantNodes);
+    }
+
+    /**
+     * Find all values the specified dimension can have.
+     *
+     * Example::
+     *
+     *     Neos.Dimension.findPotentialDimensionValues(node, 'language')
+     *
+     * @param Node $node
+     * @param string|ContentDimensionId $dimensionName
+     * @return ContentDimensionValues
+     */
+    public function findPotentialDimensionValues(Node $node, string|ContentDimensionId $dimensionName): ContentDimensionValues
```

Comment:
should we really pass a node in here? :) same argumentation as before.

Reply:
=> find weg

Reply:
=> weg.

Reply:
ToDo: Remove because we don't need it at this point. Maybe we will include it in the future.